### PR TITLE
[TAS-2093] 🚸 Add tipping info to shopping cart item

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2024-08-28 晚上11 35 48
截圖 2024-08-28 晚上11 35 20

### CODE REVIEW

File: src/components/ShoppingCart/ListRow.vue [hunk 1]
```diff
@@ -153,6 +164,10 @@ export default {
       type: Number,
       default: 0,
     },
+    productPrice: {
+      type: Number,
+      default: 0,
+    },
```
Comment (williamchong) [inline src/components/ShoppingCart/ListRow.vue:170]: I think `productPrice` is already defined in `nft-or-collection` mixin and no need to be defined as prop

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/components/ShoppingCart/ListRow.vue [hunk 2]
```diff
@@ -173,6 +184,12 @@ export default {
         ? `${this.$t('cart_purchase_channel')} : ${this.from}`
         : '';
     },
+    tippingAmount() {
+      if (this.productPrice < this.customPrice) {
```
Comment (williamchong) [inline src/components/ShoppingCart/ListRow.vue:188]: Use `Math.max(this.customPrice - this.productPrice, 0)`

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/components/ShoppingCart/ListRow.vue (hunk 2 above)
Comment (williamchong) [reply src/components/ShoppingCart/ListRow.vue:188]: @AuroraHuang22 I mean we don't need to use `if` when we use math.max

Comment (AuroraHuang22) [reply src/components/ShoppingCart/ListRow.vue:188]: [🎨 Handle negative values](https://github.com/likecoin/liker-land/pull/1750/commits/149ccadb556e08ff9b9454b21fe2b6d5eaa42417)

Comment (AuroraHuang22) [reply src/components/ShoppingCart/ListRow.vue]: [🎨 Simplify tippingAmount calculation](https://github.com/likecoin/liker-land/pull/1750/commits/023549c178d669e40436759ea74a47968db570e1)

**Review by williamchong — APPROVED** [review]

File: src/locales/en.json [hunk 3]
```diff
@@ -89,6 +89,7 @@
   "cart_purchase_channel": "channel",
   "cart_item_added": "Item added to cart",
   "cart_item_free_not_supported": "Free items cannot be added to cart",
+  "cart_item_tipping": "Price includes a tip {amount} USD",
```
Comment (nwingt) [inline src/locales/en.json:92]:
```suggestion
  "cart_item_tipping": "Price includes a tip of {amount} USD",

```

Comment (AuroraHuang22) [reply src/locales/en.json]: [💬 Update wording](https://github.com/likecoin/liker-land/pull/1750/commits/93f789dc3484a1fc48dc42cc9edea69d3309439b)

**Review by nwingt — APPROVED** [review]

**Pull request merged**